catt/cli.py

```diff
@@ -103,28 +107,111 @@ def write_config(settings):
+            return str(entry_path.resolve())
+    return None
+
+
+def convert_srt_to_webvtt_helper(content):
+    content = re.sub(r'([\d]+)\,([\d]+)', r'\1.\2', content)
+
+    with tempfile.NamedTemporaryFile(mode='w+b',
+                                     suffix=".vtt",
+                                     delete=False) as vttfile:
+        target_filename = vttfile.name
+        vttfile.write("WEBVTT\n\n".encode())
+        vttfile.write(content.encode())
+        return target_filename
+
+
+def convert_srt_to_webvtt(filename):
+    # print("Converting {} to WebVTT".format(filename))
+    for possible_encoding in ['utf-8', 'iso-8859-15']:
+        try:
+            with open(filename, 'r', encoding=possible_encoding) as srtfile:
+                content = srtfile.read()
+                return convert_srt_to_webvtt_helper(content)
+        except UnicodeDecodeError:
+            pass
+    raise CattCliError("Could not find the proper encoding of {}. Please convert it to utf-8".format(filename))
+
+
+def load_subtitle_if_exists(subtitle, video, local_ip, port):
+    subtitle = subtitle or hunt_subtitle(video)
+    if subtitle is None:
+        return None
+    click.echo("Using subtitle {}".format(subtitle))
+
+    if "://" in subtitle:
+        # it's an URL
+        if subtitle.lower().endswith(".srt"):
+            content = requests.get(subtitle).text
+            subtitle = convert_srt_to_webvtt_helper(content)
+        else:
+            return subtitle
+
+    if subtitle.lower().endswith(".srt"):
+        subtitle = convert_srt_to_webvtt(subtitle)
+
+    thr = Thread(target=serve_file,
+                 args=(subtitle, local_ip, port, "text/vtt;charset=utf-8"))
+    thr.setDaemon(True)
+    thr.start()
+    subtitle_url = "http://{}:{}/{}".format(local_ip, port, subtitle)
+    return subtitle_url
+
+
+def process_subtitle(ctx, param, value):
+    if value is None:
+        return None
+
+    if "://" in value:
+        return value
+
+    if not Path(value).is_file():
+        raise CattCliError("Subtitle file [{}] does not exist".format(value))
+
+    return value
+
+
 @cli.command(short_help="Send a video to a Chromecast for playing.")
 @click.argument("video_url", callback=process_url)
+@click.argument("subtitle", required=False, callback=process_subtitle)
 @click.option("-f", "--force-default", is_flag=True,
               help="Force use of the default Chromecast app (use if a custom app doesn't work).")
 @click.option("-r", "--random-play", is_flag=True,
               help="Play random item from playlist, if applicable.")
+@click.option("--dont-load-subtitles-automatically", is_flag=True, default=False,
+              help="Don't try to load subtitles automatically from the local folder.")
 @click.pass_obj
-def cast(settings, video_url, force_default, random_play):
+def cast(settings, video_url, subtitle, force_default, random_play, dont_load_subtitles_automatically):
```

Comment:
Having `subtitle` be an option would work better here as well, as I don't like changing function signatures that drastically. However, we don't have default arguments for this, so it'd break anyway.

Reply:
Oh, I misread "subtitle without any argument". I think loading the associated sub should be the default behaviour, without needing an argument...

Reply:
I.e. without needing to specify `--subtitle`, I mean.